src/null-safety/faq.md

```diff
@@ -319,72 +319,75 @@ and situations where a null value is really expected. So the tool tells you what
 it knows ("it looks like this condition will always be false!") and lets you
 decide what to do.
 
-## What should I know about dart2js and null safety?
+## What should I know about compiling and null safety?
 
-For a long time, the dart2js compiler has had optimizations specifically
-targeting null values and null checks. Because of that, null safety is not
-expected to affect much the output of the compiler.
+The compiler optimizes null values and null checks. So null safety
+should not affect the compiler output much.
 
 A few notes that are worth highlighting:
 
-* dart2js emits `!` null assertions, but you may not notice them.
-  That's because pre null-safety dart2js already emitted null checks
-  (so they are already in the existing programs).
-
-* dart2js emits these null assertions regardless of sound/unsound null safety
-  and regardless of optimization level. In fact, dart2js doesn't remove
-  `!` when using `-O3` or `--omit-implicit-checks`.
-
-* dart2js may optimize away unnecessary null checks. This is because the same
-  optimizations done by dart2js in the past will be able to eliminate
-  unnecessary null checks when it knows the value is not null.
-
-* By default dart2js would generate parameter subtype checks (runtime checks
-  used to ensure covariant virtual calls are given appropriate arguments).
-  These are elided with the `--omit-implicit-checks` flag just as before.
-  Recall that this flag can make programs have unexpected behavior if types
-  are invalid, so we continue to recommend that the code has strong test
-  coverage to avoid any surprises. In particular, dart2js optimizes code based
+* The compiler generates `!` null assertions, but you might not notice
+  them. Pre null-safety, the compiler generated null checks that still
+  remain in existing programs).
+
+* The compiler generates these null assertions regardless of the
+  soundness of null safety or optimization level. In fact, the compiler
+  doesn't remove `!` when using `-O3` or `--omit-implicit-checks`.
+
+* The compiler might remove unnecessary null checks. Previous compiler
```

Comment:
I have a similar concern here as with the the intro paragraph: I think the context is getting lost. Seeing your edit reveal to me that the original wording was vague, so it highlights that we need to be more explicit about that context.

A suggestion here may be to clarify that when we used to say "optimizations done in the past" we meant "optimizations done in the compiler prior to null-safety".